### ISSUE TITLE
Catch errors related to the expiration of SSL certification reporting them gracefully

### ISSUE THREAD

**Issue by pdelfino** [atlas-engineer/nyxt#2075]
**Is your feature request related to a problem? Please describe.**

Yes this feature request is related to a problem. I am using NixOS and there was a configuration problem (already fixed) with the expiration of SSL certificates. This would be clear while using the library Dexador (which is heavily used by Nyxt). 

For instance:
```lisp
CL-USER> (dex:get "http://lisp.org/")

SSL verify error: 10 X509_V_ERR_CERT_HAS_EXPIRED
```
**Describe the solution you'd like**

Nyxt should catch and report this _class of_ error **gracefully**.

**Describe alternatives you've considered**
I cannot think of an alternative.

**Additional context**
This enhancement was suggested by @Ambrevar on a [discussion](https://github.com/atlas-engineer/demeter/issues/46#issuecomment-1027671847) of a bug report in [Demeter](https://nyxt.atlas.engineer/application/demeter.org)'s private repo.

**Comment by guojing0** [comment]
> Nyxt should catch and report this class of error gracefully.

For normal users (non-developers), by "report" do you mean to have some page/buffer in Nyxt that would display the SSL problem to the users, like a 404 page?

**Comment by pdelfino** [comment]
This was actually @Ambrevar' suggestion in Demeter's private repo. I think it is better to let him make it clear.

**Comment by Ambrevar** [comment]
Yes, maybe a new internal page showing the expired certificate, à-la Firefox.

**Comment by jmercouris** [comment]
The certificates used for Dexador and WebKitGTK+ can be different. It may be possible that Lisp has access to an expired certificate, whereas GTK within the same exact binary can request things just fine.

**Comment by aadcg** [comment]
I believe this can be closed after f66a29ce09c4b72a26a6ef1a590544b1b5ebc49a.
